model/project_parser_db.go

```diff
@@ -53,64 +57,111 @@ func ParserProjectFindOneById(id string) (*ParserProject, error) {
 }
 
 // UpdateOne updates one project
-func ParserProjectUpdateOne(query interface{}, update interface{}) error {
-	return db.Update(
+func ParserProjectUpsertOne(query interface{}, update interface{}) error {
+	_, err := db.Upsert(
 		ParserProjectCollection,
 		query,
 		update,
 	)
+
+	return err
+}
+
+func setAllFieldsUpdate(pp *ParserProject) interface{} {
+	return bson.M{
+		"$set": bson.M{
+			ParserProjectEnabledKey:           pp.Enabled,
+			ParserProjectStepbackKey:          pp.Stepback,
+			ParserProjectPreErrorFailsTaskKey: pp.PreErrorFailsTask,
+			ParserProjectBatchTimeKey:         pp.BatchTime,
+			ParserProjectOwnerKey:             pp.Owner,
+			ParserProjectRepoKey:              pp.Repo,
+			ParserProjectRepoKindKey:          pp.RepoKind,
+			ParserProjectRemotePathKey:        pp.RemotePath,
+			ParserProjectBranchKey:            pp.Branch,
+			ParserProjectIdentifierKey:        pp.Identifier,
+			ParserProjectDisplayNameKey:       pp.DisplayName,
+			ParserProjectCommandTypeKey:       pp.CommandType,
+			ParserProjectIgnoreKey:            pp.Ignore,
+			ParserProjectPreKey:               pp.Pre,
+			ParserProjectPostKey:              pp.Post,
+			ParserProjectTimeoutKey:           pp.Timeout,
+			ParserProjectCallbackTimeoutKey:   pp.CallbackTimeout,
+			ParserProjectModulesKey:           pp.Modules,
+			ParserProjectBuildVariantsKey:     pp.BuildVariants,
+			ParserProjectFunctionsKey:         pp.Functions,
+			ParserProjectTaskGroupsKey:        pp.TaskGroups,
+			ParserProjectTasksKey:             pp.Tasks,
+			ParserProjectExecTimeoutSecsKey:   pp.ExecTimeoutSecs,
+			ParserProjectLoggersKey:           pp.Loggers,
+			ParserProjectAxesKey:              pp.Axes,
+			ParserProjectConfigNumberKey:      pp.ConfigUpdateNumber,
+		},
+	}
+}
+
+func checkConfigNumberQuery(id string, configNum int) bson.M {
+	q := bson.M{ParserProjectIdKey: id}
+	if configNum == 0 {
+		q["$or"] = []bson.M{
+			bson.M{ParserProjectConfigNumberKey: bson.M{"$exists": false}},
+			bson.M{ParserProjectConfigNumberKey: configNum},
```

Comment:
Shouldn't the query be this instead? (since ParserProjectConfigNumberKey is omitempty)
```
if configNum == 0 {
  q[ParserProjectConfigNumberKey] = bson.M{"$exists": false}
} else {
  q[ParserProjectConfigNumberKey] = configNum
}
```